src/lib/checkers.js

```diff
@@ -1,17 +1,42 @@
 /* eslint-disable no-console */
+'use strict';
 
+/**
+ * Find users who currently belong to the organization, but not who are not
+ * expected by config.
+ *
+ * @param {string[]} configured - An array of usernames expected to exist in the org
+ * @param {string[]} retrieved - An array of usernames actually belong to the org currently
+ * @returns {string[]} An array of usernames which are not expected to belong to the org
+ */
 function findUndocumentedMembers(configured, retrieved) {
   const undocumented = retrieved.filter(member => !configured.includes(member));
   console.log(`${undocumented.length} undocumented memberships found: `, undocumented);
   return undocumented;
 }
 
+/**
+ * Find users who are expected by config, but who do not (yet) belong to the
+ * organization.
+ *
+ * @param {string[]} configured - An array of usernames expected to exist in the org
+ * @param {string[]} retrieved - An array of usernames actually belong to the org currently
+ * @returns {string[]} An array of usernames which are expected to belong to the org
+ */
 function findNewMembers(configured, retrieved) {
   const newMembers = configured.filter(member => !retrieved.includes(member));
   console.log(`${newMembers.length} new memberships requested: `, newMembers);
   return newMembers;
 }
 
+/**
+ * Find users who are currently admins of the org, but should not be, according
+ * to config.
+ *
+ * @param {object.<string, string>} configured - Configured usernames and their associated roles
+ * @param {object.<string, object.<string, string>>} retrieved - User list retrieved from GitHub
+ * @returns {string[]} An array of usernames who are admins, but should not be
+ */
 function findDemotions(configured, retrieved) {
```

Comment:
This PR brings to light that you have functions that reuse the same names for things but what they are are actually different. IMHO consider naming these more specifically so it's more obvious if you want just the array of usernames or these deeper objects (or just make everything work off the same deeper objects)

Reply:
Yeah... I noticed the same thing. It made me start to cringe a bit. I'm starting to wonder if it's worth rethinking what actual data these functions take in.